### PR TITLE
Week 2 homework Hikmet Erdil

### PR DESCRIPTION
**Pull request by hikmeterdil**

### CODE REVIEW

**Review by nudge — COMMENTED** [review]
Hi Hikmet! Next time remember to checkout `master` branch before creating a new branch and  pull request, otherwise last weeks changes will be in here also. Thanks!

File: Week2/homework/js-exercises/convertGrades.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            gradeUsa = 'E';
+        }
+        else if (0 <= score && score <=49) {
+            gradeUsa = 'F';
+        }
+        
+
+
+
+return console.log('You got a '  + gradeUsa + ' (' + score + '%)!');;
```
Comment (nudge) [inline Week2/homework/js-exercises/convertGrades.js:30]: You don't need to return here, as `console.log` will always return `undefined`. It is enough just to do the console log and you can remove the `return` keyword.

File: Week2/homework/js-exercises/convertGrades.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+let gradeUsa = '';
+
+let score ='';
```
Comment (nudge) [inline Week2/homework/js-exercises/convertGrades.js:5]: You can move `gradeUsa` inside the body of the `function`. This keeps the variables encapsulated inside which makes the function much more reusable. Also, you do not need to declare `score` at all -- the argument `score` already is visible inside the body of the function.

File: Week2/homework/js-exercises/drinkTray.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+const drinkTypes = ['cola', 'lemonade', 'water'];
+
+let drinkTray = [];
+
+for( let i=0;i<5;i++) {
```
Comment (nudge) [inline Week2/homework/js-exercises/drinkTray.js:7]: nit: spacing

File: Week2/homework/js-exercises/drinkTray.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+        drinkTray.push(drinkTypes[i%drinkTypes.length]);
+    
+    }
+    console.log("Hey guys, I brought a " + drinkTray + ".");
```
Comment (nudge) [inline Week2/homework/js-exercises/drinkTray.js:11]: 👍 .
Also try think of a way to do this without using the `%` operator, instead think about how to check for duplicate drinks in the `drinkTray` and add another drink if there are already 2 inside.

File: Week2/homework/js-exercises/oddNumbers.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+let numberX;
+
+for(numberX = 0; numberX <= 20; numberX++) {
+    if(numberX % 2 === 0){
+    console.log('The number ' + numberX  + ' is even');
+    }
+        else{console.log('The number ' + numberX  + ' is odd');}
+}
```
Comment (nudge) [inline Week2/homework/js-exercises/oddNumbers.js:9]: 👍 nice!

nit: mind the code formatting, it is much more readable to keep consistent spacing, something like:
```
for (numberX = 0; numberX <= 20; numberX++) {
  if (numberX % 2 === 0) {
    console.log('The number ' + numberX  + ' is even');
  } else {
    console.log('The number ' + numberX  + ' is odd');
  }
} 
```

File: Week2/homework/js-exercises/recipeLoop.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+let pizzaRecipe = {
+   'Meal Name' : 'Pizza',
+   'Servings'  : 4,
+    'Ingredients' : ['Dough','Sauce','Cheese']
+   }
+
+   for (let key in pizzaRecipe) {
+        if (pizzaRecipe.hasOwnProperty(key)) {
```
Comment (nudge) [inline Week2/homework/js-exercises/recipeLoop.js:10]: Nice, you will learn more about `hasOwnProperty` in Javascript 3 i believe.

File: Week2/homework/js-exercises/recipeLoop.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+        console.log(key + " -> " + pizzaRecipe[key]);
+    }
+       console.log()
```
Comment (nudge) [inline Week2/homework/js-exercises/recipeLoop.js:13]: nit: ^ you can remove this line

File: Week2/homework/js-exercises/w2ReplaceCommas.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+let myString2 = 'hello,this,is,a,difficult,to,read,sentence';
+
+let myString3 = "";
+for (let i = 0; i < myString2.length; i++) {
+    if(myString2[i] !== ',') {
+       myString3 += myString2[i];
+    }
+        else{myString3 += ' '}
+}
```
Comment (nudge) [inline Week2/homework/js-exercises/w2ReplaceCommas.js:18]: Interesting solution ;)

You can also use `.split()` and `.join()` functions on `String`. You can look up MDN on how to use them.

```
let newString = myString.split(",").join(" ");
```

File: Week2/homework/js-exercises/w2ReplaceCommas.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+let myString = 'hello,this,is,a,difficult,to,read,sentence';
+let newString = myString.replace(/,/g, ' ');
+
+console.log( newString );
```
Comment (nudge) [inline Week2/homework/js-exercises/w2ReplaceCommas.js:5]: :+1:

File: Week2/homework/js-exercises/bookArrayLoop.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+   let alreadyReadBooks = books[i].alreadyRead;
+   if (alreadyReadBooks === true) {
+       console.log(titleBooks + ' by ' + authorBooks);
+      console.log('You already read ' +titleBooks); 
+   }
+      else { console.log(titleBooks + ' by ' + authorBooks);
+      console.log('You still need to read ' + titleBooks);
+   }
+}
+
```
Comment (nudge) [inline Week2/homework/js-exercises/bookArrayLoop.js:20]: 👍  nice!

nit: please mind your code formatting, try keep spaces, line spacing and indents as consistent as possible to make it easier to read :)

File: Week2/homework/js-exercises/convertGrades.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+//Created this if else function to compare score with different cases to find out USA equivelent of our score.
+function gradeCalculator(score) {
+    if (90 <= score && score <= 100) {
+    gradeUsa = 'A';
+    }
+        else if (80 <= score && score <= 89) {
```
Comment (nudge) [inline Week2/homework/js-exercises/convertGrades.js:11]: best to use < 90 on the upper bound, as with your logic, a number like `89.95` would not match any condition!

**Comment by wouterkleijn** [comment]
Thanks for reviewing @nudge. Do you want @hikmeterdil to implement some of the suggestions?

**Pull request closed without merging**